main/src/cgeo/geocaching/connector/gc/GCConstants.java

```diff
@@ -80,7 +80,7 @@
     static final Pattern PATTERN_CACHES_FOUND = Pattern.compile(Pattern.quote("cache-count\">") + "\\s*([\\d,.]+)", Pattern.DOTALL);
 
     static final Pattern PATTERN_AVATAR_IMAGE_SERVERPARAMETERS = Pattern.compile("\"(https?:\\/\\/(img(?:cdn)?\\.geocaching\\.com|[^>\\\"]+\\.cloudfront\\.net)\\/avatar\\/[0-9a-f-]+\\.(" + IMAGE_FORMATS + "))\\\"");
-    static final Pattern PATTERN_LOGIN_NAME_LOGIN_PAGE = Pattern.compile("ctl00_ContentBody_lbUsername\">.*<strong>(.*)</strong>");
+    static final Pattern PATTERN_LOGIN_NAME_LOGIN_PAGE = Pattern.compile("<script type=\"text\\/javascript\">\\s*var dataLayer = \\[\\{\\s*\"username\": \"([^\"]*)\"");
```

Comment:
I wonder if the escaping of special characters (especially `"`) also changed and requires some more changes on our end.

Reply:
Might very well be. Currently I'm taking all characters except quotation mark, regardless of any specific encoding. Need to investigate more deeply for special characters which would have to be encoded in Javascript. Do you have a regexp or utility function for decoding encoded chars in JS strings?

Reply:
Additionally, the two regexpes are a bit fragile as they are position dependent within the script block - needs to be a bit more generalized. But that's a different task for another day...

Reply:
We could match the whole JSON string, then construct a `JSONObject` out of it and read the fields we need. Shouldn't be that complicated. For the JSON part, you can take a look at `main/src/cgeo/geocaching/models/CalcState.java`.

Reply:
@UniQP Can JSoup parse Javascript vars in the source, some code like this?
```
<script type="text/javascript">
//<![CDATA[
var a = true;
window['someName'] = {
  "var1": true,
  "var2": "some text",
  "var3": "",
  ...
}; window['someOtherName'] = 'someValue';//]]>
</script>
```
I would like to receive the contents of the "someName" var as key/value pairs.

Reply:
@moving-bits As JSoup is an HTML parser, I don't think it's possible (because it involves parsing JavaScript).